### DISCUSSION TITLE
表格组件中的输入控件输入时导致表格二次渲染从而丢失鼠标焦点

### DISCUSSION THREAD

**Discussion by NexxLuo** [alibaba/formily#2217] [Q&A 2.x] [answered]
自定义了一个ArrayTable组件，代码完全是copy自@formily/antd。
一旦在代码中对`clone(field.value)`，即便clone后的数据没有使用，都会导致输入时表格二次渲染，鼠标丢失焦点。
起初以为是clone方法改变了value引用，改用`JSON.parse(JSON.stringfy(field.value))`，但结果是一样的。
很奇怪...

codesandbox:

https://codesandbox.io/s/biao-ge-lie-shou-ci-bian-ji-shi-hui-duo-xuan-ran-yi-ci-biao-ge-8phf0?file=/Table.tsx

**Comment by janryWang** [comment, marked as answer]
这个是你没理解reactive机制 ，深度拷贝会触发完整数据结构的依赖收集，任何变化都会触发组件重绘
